pkg/scheduler/controller/shoot/reconciler.go

```diff
@@ -208,6 +215,33 @@ func filterSeedsMatchingProviders(cloudProfile *gardencorev1beta1.CloudProfile,
 	return matchingSeeds, nil
 }
 
+// filterSeedsMatchingMultiZonalAttribute filters seeds by label `seed.gardener.cloud/multi-zonal`,
+// based on whether the shoot requires a multi-zonal control plane or not
+func filterSeedsMatchingMultiZonalAttribute(seedList []gardencorev1beta1.Seed, shoot *gardencorev1beta1.Shoot) ([]gardencorev1beta1.Seed, error) {
+	if !gardenerschedulerfeatures.FeatureGate.Enabled(features.HAControlPlanes) {
+		return seedList, nil
+	}
+
+	if shoot.ObjectMeta.Annotations[v1beta1constants.ShootAlphaControlPlaneHighAvailability] != v1beta1constants.ShootAlphaControlPlaneHighAvailabilityMultiZone {
+		return seedList, nil
+	}
+
+	// TODO(shreyas-s-rao): switch from shoot annotation to shoot spec field for beta release of HAControlPlanes
```

Comment:
Doesn't this TODO belong to 225? Also, why was it added here? I don't see similar TODOs at other usage places of the alpha annotation. Remove it?